algorithms/searching/linear_search_recursion.c

```diff
@@ -0,0 +1,35 @@
+// C Program to implement recursive Linear Search 
+
+#include<stdio.h>
+#include<stdlib.h>
+/* Recursive function to search an element in arr[l..r] */
+int linearSearch(int arr[], int l, int r, int x)
```

Comment:
This entire function along with the comments looks very similar to the one in the link mentioned.